hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -226,7 +238,8 @@ public static Writable avroToArrayWritable(Object value, Schema schema) {
           mapValues[1] = avroToArrayWritable(mapEntry.getValue(), schema.getValueType());
           values3[index3++] = new ArrayWritable(Writable.class, mapValues);
         }
-        return new ArrayWritable(Writable.class, values3);
+        wrapperWritable = new Writable[]{new ArrayWritable(Writable.class, values3)};
```

Comment:
Hive 1.2.1 and cdh 1.1.0 hive seem to return ArrayWritables for maps/array differently. Esp, apache version has a wrapper to hold the elements. This was causing mismatch with our code, which was expecting the map items to be just top level for e.g

Meta point here is, I think we should fork the Hive Parqeut Record Reader for the Realtime Record Reader usage ourselves and maintain it going forward.. This is consistent with our goal that RO view will continue to offer the native query perf of the engine, while RT view perf is something owned by hoodie.

What do you both think? If we are in agreement, I ll go ahead and get that rolling as well. Otherwise, once we land this the cdh build would fail .

Reply:
Just like we discussed, may be we should decide if we even want to support RealTime in Hive (Hive on Spark).